### PR TITLE
Fix for #3607 and #3515

### PR DESCRIPTION
**Pull request by BaronGreenback**
Fixes for:

https://github.com/jellyfin/jellyfin/issues/3607
- code could be improved - but I've focused on a minimal change due to limited testing - and that this code will hopefully be gone in the near future. 

Includes fix for breaking change, reverting back the breaking change previously implemented.
- SSDP: Include any address if LAN subnets aren't specified #3515

### CODE REVIEW

File: Emby.Server.Implementations/Networking/NetworkManager.cs [hunk 1]
```diff
@@ -152,6 +152,11 @@ private bool IsInPrivateAddressSpace(string endpoint, bool checkSubnets)
                 return true;
             }
 
+            if (!IPAddress.TryParse(endpoint, out _))
```
Comment (BaronGreenback) [inline Emby.Server.Implementations/Networking/NetworkManager.cs:155]: IP address validation to stop it from crashing @ line 160.

File: Emby.Server.Implementations/Networking/NetworkManager.cs [hunk 2]
```diff
@@ -268,6 +273,12 @@ public bool IsAddressInSubnets(IPAddress address, bool excludeInterfaces, bool e
             string excludeAddress = "[" + addressString + "]";
             var subnets = LocalSubnetsFn();
 
+            // Include any address if LAN subnets aren't specified
```
Comment (BaronGreenback) [inline Emby.Server.Implementations/Networking/NetworkManager.cs:276]: Re-instates 10.5 functionality. If the LAN subnets are empty - all addresses are deemed to be a local address. (see #3515 for conversation around this).

File: Emby.Server.Implementations/Networking/NetworkManager.cs [hunk 3]
```diff
@@ -152,6 +152,11 @@ private bool IsInPrivateAddressSpace(string endpoint, bool checkSubnets)
                 return true;
             }
 
+            if (!IPAddress.TryParse(endpoint, out _))
+            {
+                return false;
+            }
+
             byte[] octet = IPAddress.Parse(endpoint).GetAddressBytes();
```
Comment (crobibero) [inline Emby.Server.Implementations/Networking/NetworkManager.cs:160]:
```suggestion
            if (!IPAddress.TryParse(endpoint, out var ipAddress))
            {
                return false;
            }

            byte[] octet = ipAddress.GetAddressBytes();
```

Don't need to parse the ip twice.

Comment (BaronGreenback) [reply Emby.Server.Implementations/Networking/NetworkManager.cs:160]: @Bond-009 Sorry, thought resolved was accept change.

**Pull request merged**